src/main/controllers/SupportingMaterialController.ts

```diff
@@ -12,18 +11,12 @@ const logger = getLogger('SupportingMaterialController');
  */
 export default class SupportingMaterialController {
   public async get(req: AppRequest, res: Response): Promise<void> {
-    const userCase = req.session?.userCase;
-    let fileId = userCase?.contactApplicationFile?.document_url || userCase?.supportingMaterialFile?.document_url;
-    if (!fileId && userCase.selectedGenericTseApplication) {
-      fileId = userCase.selectedGenericTseApplication.value.documentUpload.document_url;
-    }
-
+    const docId = req.params.docId;
```

Comment:
I think we need to bring back some validation on this controller. I count three use cases so far. 

- Getting the supporting material added by the respondent to the selected application
- Getting the supporting material added by the claimant in the response to the selected application
- Getting the sendNotificationUploadDocument from the selectedRequestOrOrder

Similar to the validation on Line 21 of the deleted TribunalOrderDocumentController, use the docId provided to check if the document belongs to the case. If any of the three conditions are true, continue to make the call to the CaseApi... else return not found.

Reply:
Added validation